pkg/kotsadmsnapshot/backup.go

```diff
@@ -743,63 +743,73 @@ func ListBackupsForApp(ctx context.Context, kotsadmNamespace string, appID strin
 			continue
 		}
 
-		backup := types.Backup{
-			Name:   veleroBackup.Name,
-			Status: types.GetStatusFromBackupPhase(veleroBackup.Status.Phase),
-			AppID:  appID,
+		back, err := ParseVeleroBackup(ctx, veleroBackup)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse velero backup")
 		}
 
-		if veleroBackup.Status.StartTimestamp != nil {
-			startedAt := veleroBackup.Status.StartTimestamp.Time.UTC()
-			backup.StartedAt = &startedAt
-		}
-		if veleroBackup.Status.CompletionTimestamp != nil {
-			finishedAt := veleroBackup.Status.CompletionTimestamp.Time.UTC()
-			backup.FinishedAt = &finishedAt
-		}
-		if veleroBackup.Status.Expiration != nil {
-			expiresAt := veleroBackup.Status.Expiration.Time.UTC()
-			backup.ExpiresAt = &expiresAt
+		if back != nil {
```

Comment:
Can backup be nil? What would it mean for it to be nil? If you really want to handle this case is it better to return an error.

Reply:
it looks like it can't be nil, I'll remove this